src/equations/2d/linear_scalar_advection.jl

```diff
@@ -24,18 +24,27 @@ get_name(::LinearScalarAdvectionEquation2D) = "LinearScalarAdvectionEquation2D"
 varnames_cons(::LinearScalarAdvectionEquation2D) = SVector("scalar")
 varnames_prim(::LinearScalarAdvectionEquation2D) = SVector("scalar")
 
+function x_trans_periodic(x)
+  #Fixed domain_length and center.
+  domain_length = [2,2]
+  center = [0,0]
+  x = (x - center) .% (domain_length)
+  x += ((x .<= -0.5*domain_length) .- (x .>= 0.5*domain_length)) .* (domain_length)
+  return x + center
```

Comment:
As a general note, instead of re-using `x`, it might be clearer to introduce new variables and then combine everything in the return value. E.g., the first `x = ...` could be named `x_normalized = `, the `x += ...` could be `periodic_shift = ...` and then the return statement could be something like `return center + x_normalized + periodic_shift`. But it does not have to be this way, and you also may choose better names that I suggested ;-)

Reply:
This is a good idea to create more clarity. I´ll introduce varibles names accordingly.